### PR TITLE
Issue 48: company model cleanup

### PR DESCRIPTION
**Pull request by aseemavi**
Removed Unnecessary identity additions as these are now part of base modules.
Fixed Comments.
Added QoS-group/bw-remaining to compa model.

### CODE REVIEW

File: src/yang/example-compa-diffserv.yang [hunk 1]
```diff
@@ -32,60 +32,73 @@ module example-compa-diffserv {
...
     base qos-types:filter-type;
     description
-      "qos-group filter-type";
-  }
-
-  identity drop {
-    base qos-types:action-type;
-    description
-      "drop action-type";
+      "Filter-type to match on  packet metadata.";
```
Comment (mjethanandani) [inline src/yang/example-compa-diffserv.yang:49]: First of all, the name of the identity qos-group is confusing to me. What is a qos-group? The term has not been used in the draft anywhere. Can the name reflect what the identity really is. 

I also do not understand this description. Is this an identity that defines filtering based on matches within a packet metadata? If so, it would help to reword this to:

"A filter type that matches on packet metadata."

But the larger question is, what is packet metadata? This is the first time I am hearing about packet metadata. Are these fields within a packet? If so, can we use that in the description.

File: src/yang/example-compa-diffserv.yang [hunk 2]
```diff
@@ -32,60 +32,73 @@ module example-compa-diffserv {
...
-        type empty;
+  grouping bw-remaining-percent {
+    container bw-remaining-percent {
+      leaf value {
+        type uint8;
         description
-          "drop action";
+          "Value for the rate that is configured.";
+      }
+        description
```
Comment (mjethanandani) [inline src/yang/example-compa-diffserv.yang:83]: Indentation is off.

File: src/yang/example-compa-diffserv.yang [hunk 3]
```diff
@@ -32,60 +32,73 @@ module example-compa-diffserv {
...
-    container drop {
-      leaf drop {
-        type empty;
+  grouping bw-remaining-percent {
+    container bw-remaining-percent {
+      leaf value {
+        type uint8;
         description
-          "drop action";
+          "Value for the rate that is configured.";
```
Comment (mjethanandani) [inline src/yang/example-compa-diffserv.yang:81]: What value is this? Reserved and priority bandwidth? Or what is left over? What are its units?

File: src/yang/example-compa-diffserv.yang [hunk 4]
```diff
@@ -32,60 +32,73 @@ module example-compa-diffserv {
...
+  grouping qos-group-marking {
+    container qos-group {
+      leaf qos-group {
+        type uint8;
+        description
+          "Mark metadata information in the network packet";
       }
       description
-        "drop container";
+        "Container for Qos-Group marking.";
```
Comment (mjethanandani) [inline src/yang/example-compa-diffserv.yang:98]: I am confused about what is QoS group, so once you have explained somewhere what it is, this may need an update.

Also can we be consistent in using QoS rather than Qos.

File: src/yang/example-compa-diffserv.yang [hunk 5]
```diff
@@ -114,8 +128,14 @@ module example-compa-diffserv {
...
     }
+    case bw-remaining {
+      uses bw-remaining-percent;
+    }
+    case mark-qos-group {
+      uses qos-group-marking;
+    }
     description
-      "Augment the actions to policy entry";
+      "Augment the actions to policy entry.";
```
Comment (mjethanandani) [inline src/yang/example-compa-diffserv.yang:138]: I think you mean

"Augments policy entry to add actions."

File: src/yang/example-compa-diffserv.yang [hunk 6]
```diff
@@ -129,17 +149,17 @@ module example-compa-diffserv {
     "/diffserv:single-rate-two-color-meter" {
 
     description
-      "augment the single-rate-two-color meter with" +
-      "color classifiers";
+      "Augment the single-rate-two-color meter with" +
```
Comment (mjethanandani) [inline src/yang/example-compa-diffserv.yang:152]: s/with/to add/

File: src/yang/example-compa-diffserv.yang [hunk 7]
```diff
@@ -153,24 +173,25 @@ module example-compa-diffserv {
     "/diffserv:single-rate-three-color-meter-type" +
     "/diffserv:single-rate-three-color-meter" {
     description
-      "augment the one-rate-tri-color meter with" +
-      "color classifiers";
+      "Augment the one-rate-tri-color meter with" +
```
Comment (mjethanandani) [inline src/yang/example-compa-diffserv.yang:176]: s/with/to add/

File: src/yang/example-compa-diffserv.yang [hunk 8]
```diff
@@ -181,22 +202,22 @@ module example-compa-diffserv {
     "/diffserv:two-rate-three-color-meter-type" +
     "/diffserv:two-rate-three-color-meter" {
     description
-      "augment the two-rate-tri-color meter with" +
-      "color classifiers";
+      "Augment the two-rate-tri-color meter with" +
```
Comment (mjethanandani) [inline src/yang/example-compa-diffserv.yang:205]: s/with/to add/

File: src/yang/example-compb-diffserv-filter-policy.yang [hunk 9]
```diff
@@ -45,57 +45,20 @@ module example-compb-diffserv-filter-policy {
 
   revision YYYY-MM-DD {
     description
-      "Initial revision of diffserv policy";
+      "Initial revision of diffserv policy.";
```
Comment (mjethanandani) [inline src/yang/example-compb-diffserv-filter-policy.yang:48]: How about

"Initial revision of DiffServ policy for Company B."?

File: src/yang/example-compb-diffserv-filter-policy.yang [hunk 10]
```diff
@@ -106,20 +69,20 @@ module example-compb-diffserv-filter-policy {
       leaf loss-priority {
         type enumeration {
           enum high {
-           description "High Loss Priority";
+           description "High Loss Priority.";
```
Comment (mjethanandani) [inline src/yang/example-compb-diffserv-filter-policy.yang:72]: s/Loss Priority/loss priority/

unless you are referring to a special kind of high loss priority.

File: src/yang/example-compb-diffserv-filter-policy.yang [hunk 11]
```diff
@@ -106,20 +69,20 @@ module example-compb-diffserv-filter-policy {
...
           }
           enum medium-high {
-           description "Medium-high Loss Priority";
+           description "Medium-high Loss Priority.";
```
Comment (mjethanandani) [inline src/yang/example-compb-diffserv-filter-policy.yang:75]: s/Medium-high Loss Priority/Medium to high loss priority/

File: src/yang/example-compb-diffserv-filter-policy.yang [hunk 12]
```diff
@@ -106,20 +69,20 @@ module example-compb-diffserv-filter-policy {
...
           }
           enum medium-low {
-           description "Medium-low Loss Priority";
+           description "Medium-low Loss Priority.";
```
Comment (mjethanandani) [inline src/yang/example-compb-diffserv-filter-policy.yang:78]: Something similar here.

File: src/yang/example-compb-diffserv-filter-policy.yang [hunk 13]
```diff
@@ -106,20 +69,20 @@ module example-compb-diffserv-filter-policy {
...
           }
           enum low {
-           description "Low Loss Priority";
+           description "Low Loss Priority.";
```
Comment (mjethanandani) [inline src/yang/example-compb-diffserv-filter-policy.yang:81]: And here.

File: src/yang/example-compb-diffserv-filter-policy.yang [hunk 14]
```diff
@@ -106,20 +69,20 @@ module example-compb-diffserv-filter-policy {
...
           }
         }
         description
-          "Loss-priority";
+          "Loss-priority.";
```
Comment (mjethanandani) [inline src/yang/example-compb-diffserv-filter-policy.yang:85]: s/Loss-priority/Loss priority/

File: src/yang/example-compb-diffserv-filter-policy.yang [hunk 15]
```diff
@@ -131,108 +94,84 @@ module example-compb-diffserv-filter-policy {
...
-    }
     case internal-loss-priority {
         uses loss-priority;
       description
         "Filter Type Internal-loss-priority";
     }
     description 
       "Augments Diffserv Classifier with vendor" +
-      " specific types";
+      " specific types.";
```
Comment (mjethanandani) [inline src/yang/example-compb-diffserv-filter-policy.yang:104]: s/" specific types/"specific types/

File: src/yang/example-compb-diffserv-filter-policy.yang [hunk 16]
```diff
@@ -131,108 +94,84 @@ module example-compb-diffserv-filter-policy {
...
-      "mark-fwd-class grouping";
-  }
-
   grouping mark-loss-priority {
     container mark-loss-priority {
       leaf loss-priority {
         type enumeration {
           enum high {
-           description "High Loss Priority";
+           description "High Loss Priority.";
```
Comment (mjethanandani) [inline src/yang/example-compb-diffserv-filter-policy.yang:122]: Same as above.

File: src/yang/example-compb-diffserv-filter-policy.yang [hunk 17]
```diff
@@ -131,108 +94,84 @@ module example-compb-diffserv-filter-policy {
...
           }
           enum medium-high {
-           description "Medium-high Loss Priority";
+           description "Medium-high Loss Priority.";
```
Comment (mjethanandani) [inline src/yang/example-compb-diffserv-filter-policy.yang:125]: And here.

File: src/yang/example-compb-diffserv-filter-policy.yang [hunk 18]
```diff
@@ -131,108 +94,84 @@ module example-compb-diffserv-filter-policy {
...
           }
           enum medium-low {
-           description "Medium-low Loss Priority";
+           description "Medium-low Loss Priority.";
```
Comment (mjethanandani) [inline src/yang/example-compb-diffserv-filter-policy.yang:128]: And here.

File: src/yang/example-compb-diffserv-filter-policy.yang [hunk 19]
```diff
@@ -131,108 +94,84 @@ module example-compb-diffserv-filter-policy {
...
           }
           enum low {
-           description "Low Loss Priority";
+           description "Low Loss Priority.";
```
Comment (mjethanandani) [inline src/yang/example-compb-diffserv-filter-policy.yang:131]: And here.

File: src/yang/example-compa-diffserv.yang (hunk 1 above)
Comment (aseemavi) [reply src/yang/example-compa-diffserv.yang:49]: this is cisco specific.. so we may not look much into it?

File: src/yang/example-compa-diffserv.yang (hunk 4 above)
Comment (aseemavi) [reply src/yang/example-compa-diffserv.yang:98]: QoS group is cisco specific packet metadata which can have value 0-255.

File: src/yang/example-compb-diffserv-filter-policy.yang (hunk 15 above)
Comment (aseemavi) [reply src/yang/example-compb-diffserv-filter-policy.yang:104]: in context of previous line, a space is needed

File: src/yang/example-compa-diffserv.yang (hunk 3 above)
Comment (aseemavi) [reply src/yang/example-compa-diffserv.yang:81]: what is left over. Unit is percentage

File: src/yang/example-compa-diffserv.yang (hunk 1 above)
Comment (mjethanandani) [reply src/yang/example-compa-diffserv.yang:49]: Even if it a Cisco specific example, it needs to be clear. I would rather that we spend the time to clean it up and make it clear what the example is trying to demonstrate.

File: src/yang/example-compa-diffserv.yang (hunk 4 above)
Comment (mjethanandani) [reply src/yang/example-compa-diffserv.yang:98]: Ok. Then let us explain that in the example, without mentioning Cisco of course. Also which specific data in the packet is being looked at for the value of 0-255?

File: src/yang/example-compa-diffserv.yang (hunk 3 above)
Comment (mjethanandani) [reply src/yang/example-compa-diffserv.yang:82]: Then it is not a value. It is a percentage of the remaining bandwidth.

File: src/yang/example-compa-diffserv.yang (hunk 4 above)
Comment (aseemavi) [reply src/yang/example-compa-diffserv.yang]: it is just a tag

**Review by mjethanandani — COMMENTED** [review]
Other than the single comment, I think we need to explain what is QoS-group, and what is packet metadata in this example.

Thanks for taking care of the remaining comments.

File: src/yang/example-compa-diffserv.yang (hunk 4 above)
Comment (mjethanandani) [reply src/yang/example-compa-diffserv.yang]: I assume it is a tag that is carried separately from the packet?? Let us say that somewhere in the example.

File: src/yang/example-compa-diffserv.yang (hunk 1 above)
Comment (aseemavi) [reply src/yang/example-compa-diffserv.yang:49]: Mahesh, it is same as traffic-group in our model.. traffic-group is converted into an integer internally.. QoS-group is an integer from config. Also, traffic-group is mainly used for queuing, QoS-group can be used for other things also since 256 values are possible while traffic-groups are typically 8.

File: src/yang/example-compa-diffserv.yang (hunk 4 above)
Comment (aseemavi) [reply src/yang/example-compa-diffserv.yang]: QoS group is simply set of packet tagged as QoS group.

File: src/yang/example-compa-diffserv.yang (hunk 1 above)
Comment (aseemavi) [reply src/yang/example-compa-diffserv.yang:49]: I have clarified further. Please check.

Comment (mjethanandani) [reply src/yang/example-compa-diffserv.yang]: Thanks. Can we do something similar for the description of qos-group on line 58?

File: src/yang/example-compa-diffserv.yang [hunk 20]
```diff
@@ -32,60 +32,76 @@ module example-compa-diffserv {
...
   }
 
   grouping qos-group {
     list qos-group {
-      key "qos-group-min qos-group-max";
+      key "min max";
       description
-        "list of dscp ranges";
-      leaf qos-group-min {
+        "List of qos-group ranges.";
```
Comment (mjethanandani) [inline src/yang/example-compa-diffserv.yang:58]: Something like ..

"List of qos-groups. A QoS group is a set of packet tagged as group."

**Review by mjethanandani — COMMENTED** [review]
Just a couple more comments...

**Review by mjethanandani — APPROVED** [review]
